orc8r/cloud/go/swagger/combine.go

```diff
@@ -0,0 +1,197 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger
+
+import (
+	"regexp"
+	"sort"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// Combine multiple Swagger specs, giving precedence to the "common" spec.
+// First error contains merge warnings for overwritten fields and
+// incompatible Swagger specs which will be explicitly ignored in the merging
+// process.
+// This custom-built functionality mirrors the "official" implementation.
+// See: https://github.com/go-openapi/analysis/blob/master/mixin.go
+func Combine(common string, spec []string) (string, error, error) {
```

Comment:
- spec -> specs
- throughout this file + anywhere relevant in the tools code: be consistent about the ordering of  `common` and `specs` -- e.g. `unmarshalToSwagger` takes (common, specs) but returns (specs, common)
- we also want consistent naming patterns -- note how `unmarshalToSwagger` is transforming `spec` into `specs` -- that doesn't provide useful information to the reader, and also is misleading because both the input and output in this case are plurals

For the final point: I like `yamlSpecs` -> `specs` -> `combined` -> `out` and `yamlCommon` -> `common` -> ...